File: JANUARY/11-01-2026/85.c
int maximalRectangle(char** matrix, int matrixSize, int* matrixColSize) {
    if (matrixSize == 0) return 0;

    int cols = matrixColSize[0];
    int* heights = (int*)calloc(cols, sizeof(int));
    int* stack = (int*)malloc((cols + 1) * sizeof(int));
    int ans = 0;

    for (int i = 0; i < matrixSize; i++) {
        for (int j = 0; j < cols; j++) {
            heights[j] = (matrix[i][j] == '1') ? heights[j] + 1 : 0;
        }

        int top = -1;
        for (int k = 0; k <= cols; k++) {
            int h = (k == cols) ? 0 : heights[k];
            while (top != -1 && h < heights[stack[top]]) {
                int height = heights[stack[top--]];
                int width = (top == -1) ? k : k - stack[top] - 1;
                int area = height * width;
                if (area > ans) ans = area;
            }
            stack[++top] = k;
        }
    }

    free(heights);
    free(stack);
    return ans;
}

File: JANUARY/11-01-2026/85.go
func maximalRectangle(matrix [][]byte) int {
	if len(matrix) == 0 {
		return 0
	}

	cols := len(matrix[0])
	heights := make([]int, cols)
	ans := 0

	for i := 0; i < len(matrix); i++ {
		for j := 0; j < cols; j++ {
			if matrix[i][j] == '1' {
				heights[j]++
			} else {
				heights[j] = 0
			}
		}

		stack := []int{}
		for k := 0; k <= cols; k++ {
			h := 0
			if k < cols {
				h = heights[k]
			}
			for len(stack) > 0 && h < heights[stack[len(stack)-1]] {
				height := heights[stack[len(stack)-1]]
				stack = stack[:len(stack)-1]
				width := k
				if len(stack) > 0 {
					width = k - stack[len(stack)-1] - 1
				}
				area := height * width
				if area > ans {
					ans = area
				}
			}
			stack = append(stack, k)
		}
	}

	return ans
}